cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -2670,10 +2677,30 @@ def migrate_volume(self, context, volume, host):
                            'folder': ds_info['folder']},)
             return (True, {'migration_status': 'error'})
 
-        return (True, None)
+    def _migrate_attached(self, context, dest_host, volume, backing):
+        try:
+            # Create a diskless backing vm, so we can attach the
+            # backing moved in a live migration back to it
+            self._remote_api.create_backing(
+                context, dest_host, volume, create_params={
+                    CREATE_PARAM_DISK_LESS: True
+                })
+            return (True, None)
+        except Exception:
```

Comment:
Can we handle "already existing" here? Is it a problem, if the shadow-VM already exist, but we have a disk on it already? Does Nova check up front?

Reply:
> Can we handle "already existing" here?
Cinder handles one part of that (already)[https://github.com/sapcc/cinder/blob/stable/train-m3/cinder/volume/drivers/vmware/vmdk.py#L890]

Indeed, Nova would need to handle the case if the Shadow-VM already has a volume attached. I added a [comment](https://github.com/sapcc/nova/pull/209/files#r681559138)

Reply:
Ah, but not in the remote-api call. I'll fix it

Reply:
It is also handled in the remote-api call. It calls `driver._create_backing`, which does return an existing backing first.